drivers/openstack/openstack.go

```diff
@@ -736,7 +736,37 @@ func (d *Driver) waitForSSHServer() error {
 		"MachineId": d.MachineId,
 		"IP":        ip,
 	}).Debug("Waiting for the SSH server to be started...")
-	return ssh.WaitForTCP(fmt.Sprintf("%s:%d", ip, d.SSHPort))
+
+	err = ssh.WaitForTCP(fmt.Sprintf("%s:%d", ip, d.SSHPort))
```

Comment:
+1 to @ctrlrsf suggestion.  @ggiamarchi can you please update to create the generic function in `ssh` package and use it in the `openstack` driver in this PR please?  We will move the other drivers over to this new method when it is merged